src/app/system/codes/view-code/view-code.component.ts

```diff
@@ -0,0 +1,215 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormGroup, FormBuilder, Validators, FormArray } from '@angular/forms';
+import { MatDialog } from '@angular/material';
+
+/** Custom Services */
+import { SystemService } from 'app/system/system.service';
+
+/** Custom Components */
+import { DeleteDialogComponent } from 'app/shared/delete-dialog/delete-dialog.component';
+
+@Component({
+  selector: 'mifosx-view-code',
+  templateUrl: './view-code.component.html',
+  styleUrls: ['./view-code.component.scss']
+})
+
+/**
+ * View Code Component.
+ */
+export class ViewCodeComponent implements OnInit {
+
+  /** Code Data */
+  codeData: any;
+  /** Code Values Data */
+  codeValuesData: any[];
+  /** Code Values Form */
+  codeValuesForm: FormGroup;
+  /** Code Value Row Status */
+  codeValueRowStatus: string[] = [];
+
+  /**
+   * Retrieves the codes and code values data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {SystemService} systemService System Service.
+   * @param {Router} router Router for navigation.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {MatDialog} dialog Dialog reference.
+   */
+  constructor(private route: ActivatedRoute,
+              private systemService: SystemService,
+              private router: Router,
+              private formBuilder: FormBuilder,
+              private dialog: MatDialog) {
+    this.route.data.subscribe((data: { code: any, codevalues: any }) => {
+      this.codeData = data.code;
+      this.codeValuesData = data.codevalues;
+    });
+  }
+
+  /**
+   * Creates and initializes the code values form.
+   */
+  ngOnInit() {
+    this.createCodeValuesForm();
+    this.initCodeValuesForm();
+  }
+
+  /**
+   * Initializes the code values form.
+   */
+  initCodeValuesForm() {
+    this.codeValuesData.forEach((codeValue: any) => {
+      this.codevalues.push(this.createCodeValuesRow(codeValue));
+      this.codeValueRowStatus.push('disabled');
+    });
+  }
+
+  /**
+   * Gets the code values form array.
+   * @returns {FormArray} Code values form array.
+   */
+  get codevalues(): FormArray {
+    return this.codeValuesForm.get('codevalues') as FormArray;
+  }
+
+  /**
+   * Creates the code values form.
+   */
+  createCodeValuesForm() {
+    this.codeValuesForm = this.formBuilder.group({
+      'codevalues': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Adds a row to the code values form.
+   */
+  addCodeValueRow() {
+    this.codevalues.push(this.createCodeValuesRow());
+    this.codevalues.at(this.codevalues.length - 1).enable();
+    this.codeValueRowStatus.push('new');
+  }
+
+  /**
+   * Creates a code value row in code values form.
+   * @param {string} codeValueName Code value name.
```

Comment:
Don't express the comment like this. It is misleading as it seems we need to pass 4 parameters.